pkg/counter/cache.go

```diff
@@ -263,13 +293,17 @@ func initializeFromResources(cl client.Client, namespace string) error {
 	}
 	reset()
 	for _, usersignup := range usersignups.Items {
-		if value, exists := usersignup.Annotations[toolchainv1alpha1.UserSignupActivationCounterAnnotationKey]; exists {
-			_, err := strconv.Atoi(value) // let's make sure the value is actually an integer
+		activations, activationsExists := usersignup.Annotations[toolchainv1alpha1.UserSignupActivationCounterAnnotationKey]
+		if activationsExists { // (DEPRECATED)
+			_, err := strconv.Atoi(activations) // let's make sure the value is actually an integer
 			if err != nil {
-				log.Error(err, "invalid number of activations", "name", usersignup.Name, "value", value)
+				log.Error(err, "invalid number of activations", "name", usersignup.Name, "value", activations)
 				continue
 			}
-			cachedCounts.UsersPerActivationCounts[value]++
+			cachedCounts.UsersPerActivationCounts[activations]++
+			emailAddress := usersignup.Annotations[toolchainv1alpha1.UserSignupUserEmailAnnotationKey]
+			domain := metrics.GetEmailDomain(emailAddress) // if email address did not exist (which should not happen), then the domain will be `external`
```

Comment:
If it should not happen, wouldn't it be fine to at least log an error in the `GetEmailDomain` function? 
Can be done in the next PR of course.

Reply:
sure, I can log an error in the `GetEmailDomain` func. 
done in 3b3a856

Reply:
Nice :+1: